### ISSUE TITLE
D415 not triggering when it should

### ISSUE THREAD

**Issue by cazador481** [charliermarsh/ruff#2994]
# Code snippet
```python
def create_and_verify_testrun_entries(
    nvregress_post_client,
    nvregress_crud_client,
    nvregress_project,
    regression_run_data=sample_regression_run_data,
    test_run_data=sample_test_run_data,
):
    """Create testrun entries in NVRegress and verify that those
    entries can be read back.

    regression_run_data is layered on top of minimal_regression_run_data
    """
    pass
```
# Expected Results:
I expect the docstring to cause a  D415 error but it isn't.
What flake8 shows
```sh
poetry run flake8 --select D415 test.py
test.py:8:1: D415 First line should end with a period, question mark, or exclamation point
```
# Actual results
```sh
 poetry run ruff --select D415 test.py

```

# poetry version
```sh
poetry run ruff --version
ruff 0.0.247
```

# pyproject.toml
```toml

[tool.ruff.pycodestyle]
max-doc-length = 120

[tool.ruff]
line-length = 120
exclude = [".venv"]
ignore = [
    "D105",
    "D107",
    "D401",
    "E501",
    "W505",
]
select = [
    "B",
    # "C",
    "D",
    "E",
    "F",
    "N",
    "W",
]

[tool.ruff.pydocstyle]
convention = "google"

[tool.ruff.per-file-ignores]
"dejavu/__init__.py" = ["F401"]
```

**Comment by charliermarsh** [comment]
This is actually an intentional deviation. We look for the first logical line, rather than the first physical line (so you should see an error if you remove the period from `back.`).

**Comment by charliermarsh** [comment]
I don't think so, but it will be documented when we add documentation for that rule specifically.
